Feat: Dropdown Filter 구현

yejineee opened the pull request:
## 구현 내용
- 필터링 드롭다운 구현
- useFilter hooks 구현
- 각 드롭다운 클릭했을 시, 클릭한 아이템 체크버튼 렌더링 구현
## 화면
![스크린샷 2020-11-11 오후 6 20 11](https://user-images.githubusercontent.com/43772082/98792939-8b689980-244a-11eb-8400-e2f782d90d8a.png)
![스크린샷 2020-11-11 오후 6 20 34](https://user-images.githubusercontent.com/43772082/98793006-9de2d300-244a-11eb-81f1-9e2b4e132697.png)
![스크린샷 2020-11-11 오후 6 20 25](https://user-images.githubusercontent.com/43772082/98793010-a0452d00-244a-11eb-9548-96e59f440c86.png)
![스크린샷 2020-11-11 오후 6 20 21](https://user-images.githubusercontent.com/43772082/98793012-a1765a00-244a-11eb-928e-1cc81fce7bd6.png)

LeeSuKyeong committed:
Feat: filter 구현 (#143)

* Feat: Filtering Dropdown 구현

필터링 버튼 클릭시 표시되는 드랍다운 구현.

* Feat: issue list filter 구현

구현 중입니다.

Commit:
Fix: userdropdown props 이름 수정

userInfo props에 user 하나만 넣어주는 것으로 수정하였다.

Commit:
Feat: useFilter hook 생성

filter에서 어느 부분을 선택했는지의 상태를 관리하고,
핸들러를 포함하고 있는 useFilter hook을 만들었다.

Commit:
Feat: 이슈 리스트 필터링  드롭다운 체크 구현

이슈 리스트 부분에서 드롭다운 체크 부분을 구현하였습니다.
label은 복수를 체크할 수 있고, 담당자, 글쓴이, 마일스톤은 하나만 체크할 수 있습니다.

Commit:
Fix: 불필요하게 반복되는 loading 확인 삭제

loading이 완료가 안되어있는지, 두 번 확인하므로,
하나는 삭젠한다.

rolled-potatoes commented on front/src/components/issue/list/header/index.js:
```diff
@@ -0,0 +1,112 @@
+                key={`filter-author-id-${user.id}`}
+                userInfo={user}
+                onClick={authorSelectHandler}
+                selected={selectedAuthorId === user.id}
```
지금 프로젝트에서는 큰 상관은 없는데 ,
이것을 외부에 따른 변수로 생성 해두면, useMemo와 useCallback 으로 최적화 할 수 있어요 !

yejineee replied:
승환님 공부할 것 주셔서 너무 감사합니다.......!

rolled-potatoes reviewed (approved):
수고하셨습니다~

Commit:
Refactor: 같은 값이면 null,  다른 값이면 새로운 값으로 업데이트

같은 값이면, null로 하고 다른 값이면 새로운 값으로 업데이트하는 로직이
같기 때문에, 유틸 함수로 만들었다. 같은 로직을 사용하는 부분은 유틸 함수로 사용하였다.

Commit:
Refactor: 중복되는 reducer 파일 삭제하고, 하나의 reducer를 사용

사이드바에서 마일스톤, 어사이니, 라벨의 리듀서 로직이 모두 같기 때문에,
하나의 리듀서를 세 곳에서 공통으로 사용하도록 하였다.
